src/_pytest/stepwise.py

```diff
@@ -47,43 +69,107 @@ def pytest_sessionfinish(session: Session) -> None:
             # Do not update cache if this process is a xdist worker to prevent
             # race conditions (#10641).
             return
-        # Clear the list of failing tests if the plugin is not active.
-        session.config.cache.set(STEPWISE_CACHE_DIR, [])
+
+
+@dataclasses.dataclass
+class StepwiseCacheInfo:
+    # The nodeid of the last failed test.
+    last_failed: str | None
+
+    # The number of tests in the last time --stepwise was run.
+    # We use this information as a simple way to invalidate the cache information, avoiding
+    # confusing behavior in case the cache is stale.
+    last_test_count: int | None
+
+    # The date when the cache was last updated, for information purposes only.
+    last_cache_date_str: str
+
+    _DATE_FORMAT: ClassVar[str] = "%Y-%m-%d %H:%M:%S"
```

Comment:
Maybe use [`datetime.isoformat()`](https://docs.python.org/3/library/datetime.html#datetime.datetime.isoformat) and [`.fromisoformat()`](https://docs.python.org/3/library/datetime.html#datetime.datetime.fromisoformat) instead, just to make it clear this is a standardized format for serialization?

Reply:
Ahh sure, thanks, not sure what I was thinking.